epoxy-adapter/src/main/java/com/airbnb/epoxy/ItemAnimatorDecorator.java

```diff
@@ -0,0 +1,256 @@
+package com.airbnb.epoxy;
+
+import android.support.annotation.CallSuper;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v7.widget.RecyclerView.ItemAnimator;
+import android.support.v7.widget.RecyclerView.State;
+import android.support.v7.widget.RecyclerView.ViewHolder;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.util.HashSet;
+import java.util.List;
+
+/**
+ * base class for decorating ItemAnimators
+ * currently only supports decorating change animations
+ */
+public class ItemAnimatorDecorator extends ItemAnimator {
+
+  private final ItemAnimator mDecorated;
+
+  // lists of holders that are currently being animated by the mDecorated ItemAnimator
+  // they may be pending or running
+  private HashSet<ViewHolder> mAppearAnims = new HashSet<>();
+  private HashSet<ViewHolder> mChangeAnims = new HashSet<>();
+  private HashSet<ViewHolder> mDisappearAnims = new HashSet<>();
+  private HashSet<ViewHolder> mPersistAnims = new HashSet<>();
+
+  public ItemAnimatorDecorator(@NonNull ItemAnimator decorated) {
+    mDecorated = decorated;
+
+    setAnimationFinishedListener(decorated, new AnimationFinishedListener() {
+      @Override
+      public void onAnimationFinished(ViewHolder holder) {
+        dispatchAnimationFinished(holder);
+
+        //remove holder from the lists of currently animating holders
+        mAppearAnims.remove(holder);
+        mChangeAnims.remove(holder);
+        mDisappearAnims.remove(holder);
+        mPersistAnims.remove(holder);
+      }
+    });
+  }
+
+  private static void setAnimationFinishedListener(ItemAnimator itemAnimator,
```

Comment:
This is only needed if we are wrapping somebody's custom item animator that doesn't extend SimpleItemAnimator, correct? Could we use a different animator if the wrapped animator does extend SimpleItemAnimator so we can avoid reflection in that common case?

That way if you're just using the built in change animation delegation to Epoxy models you won't have the reflection slow down